src/OpenFeature.Contrib.ConfigCat/ConfigCatProvider.cs

```diff
@@ -0,0 +1,99 @@
+    {
+        private const string Name = "ConfigCat Provider";
+        internal readonly IConfigCatClient Client;
+
+        /// <summary>
+        /// Creates new instance of <see cref="ConfigCatProvider"/>
+        /// </summary>
+        /// <param name="sdkKey">SDK Key to access the ConfigCat config.</param>
+        /// <param name="configBuilder">The action used to configure the client.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="sdkKey"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="sdkKey"/> is an empty string or in an invalid format.</exception>
+        public ConfigCatProvider(string sdkKey, Action<ConfigCatClientOptions> configBuilder = null)
+        {
+            Client = ConfigCatClient.Get(sdkKey, configBuilder);
+        }
+
+        /// <inheritdoc/>
+        public override Metadata GetMetadata()
+        {
+            return new Metadata(Name);
+        }
+
+        /// <inheritdoc/>
+        public override async Task<ResolutionDetails<bool>> ResolveBooleanValue(string flagKey, bool defaultValue, EvaluationContext context = null)
+        {
+            return await ProcessFlag(flagKey, context, defaultValue);
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<string>> ResolveStringValue(string flagKey, string defaultValue, EvaluationContext context = null)
+        {
+            return ProcessFlag(flagKey, context, defaultValue);
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<int>> ResolveIntegerValue(string flagKey, int defaultValue, EvaluationContext context = null)
+        {
+            return ProcessFlag(flagKey, context, defaultValue);
+        }
+
+        /// <inheritdoc/>
+        public override Task<ResolutionDetails<double>> ResolveDoubleValue(string flagKey, double defaultValue, EvaluationContext context = null)
+        {
+            return ProcessFlag(flagKey, context, defaultValue);
+        }
+
+        /// <inheritdoc/>
+        public override async Task<ResolutionDetails<Value>> ResolveStructureValue(string flagKey, Value defaultValue, EvaluationContext context = null)
+        {
+            var stringDefaultValue = defaultValue?.AsString;
+            var result = await ProcessFlag(flagKey, context, stringDefaultValue);
+            var returnValue = result.Value == null ? defaultValue : new Value(result.Value);
+            return new ResolutionDetails<Value>(flagKey, returnValue, variant: result.Variant, errorMessage: result.ErrorMessage);
+        }
+
+        private async Task<ResolutionDetails<T>> ProcessFlag<T>(string flagKey, EvaluationContext context, T defaultValue)
+        {
+            var user = context?.BuildUser();
+            var result = await Client.GetValueDetailsAsync(flagKey, defaultValue, user);
+            return string.IsNullOrEmpty(result.ErrorMessage)

```

Comment:
As the GetValueDetailsAsync handling everything  - it returns the defaultValue if needed, I'd not override this behaviour in any cases with giving back the defaultValue if the errorMessage is not empty.

I think this return can be simplified to 
`return new ResolutionDetails<T>(flagKey, result.Value, !string.IsNullOrEmpty(result.ErrorMessage) ? ParseErrorType(result.ErrorMessage) : null, variant: result.VariationId, errorMessage: result.ErrorMessage);`

Reply:
Good suggestion.